### PR TITLE
JCL-337: End-to-end test of UMA auth for Access Grants endpoints

### PR DESCRIPTION
**Pull request by NSeydoux**
This adds tests to verify that the expected authentication mechanism (UMA or OpenId) gets used when accessing the Access Grants endpoints (e.g. `/issue`)

### CODE REVIEW

File: openid/src/main/java/com/inrupt/client/openid/OpenIdSession.java [hunk 1]
```diff
@@ -225,7 +225,7 @@ public Optional<Credential> fromCache(final Request request) {
         final Credential c = credential.get();
         if (!hasExpired(c) && request != null && requestCache.get(cacheKey(request.uri())) != null) {
             LOGGER.debug("Using cached token for request: {}", request.uri());
-            return Optional.of(c);
+            return Optional.of(requestCache.get(cacheKey(request.uri())));
```
Comment (acoburn) [inline openid/src/main/java/com/inrupt/client/openid/OpenIdSession.java:228]: We'll want to check whether this cached credential has expired. This function would look something like:

```java
if (request != null) {
    final Credential c = requestCache.get(cacheKey(request.uri()));
    if (!hasExpired(c)) {
        return Optional.of(c);
    }
}
return Optional.empty();
```

File: openid/src/main/java/com/inrupt/client/openid/OpenIdSession.java [hunk 2]
```diff
@@ -239,19 +239,25 @@ public void reset() {
...
-        final Optional<Credential> cred = getCredential(ID_TOKEN, null);
-        if (cred.isPresent() && request != null) {
-            LOGGER.debug("Setting cache entry for request: {}", request.uri());
-            requestCache.put(cacheKey(request.uri()), Boolean.TRUE);
-        }
-        return CompletableFuture.completedFuture(cred);
+        return auth.authenticate(this, request, algorithms)
+                .thenApply(credential -> {
+                    if (credential != null) {
+                        requestCache.put(request.uri(), credential);
```
Comment (acoburn) [inline openid/src/main/java/com/inrupt/client/openid/OpenIdSession.java:245]: Be sure to wrap this `request.uri()` in a `cacheKey` method: `cacheKey(request.uri())`

File: openid/src/main/java/com/inrupt/client/openid/OpenIdSession.java [hunk 3]
```diff
@@ -239,19 +239,25 @@ public void reset() {
...
 
     /* deprecated */
     @Override
     public CompletionStage<Optional<Credential>> authenticate(final Request request,
             final Set<String> algorithms) {
-        return authenticate(null, request, algorithms);
+        final Optional<Credential> cred = getCredential(ID_TOKEN, null);
+        if (cred.isPresent() && request != null) {
+            LOGGER.debug("Setting cache entry for request: {}", request.uri());
+            requestCache.put(request.uri(), cred.get());
```
Comment (acoburn) [inline openid/src/main/java/com/inrupt/client/openid/OpenIdSession.java:258]:
```suggestion
            requestCache.put(cacheKey(request.uri()), cred.get());
```

File: openid/src/main/java/com/inrupt/client/openid/OpenIdSession.java (hunk 1 above)
Comment (NSeydoux) [reply openid/src/main/java/com/inrupt/client/openid/OpenIdSession.java]: Fixed in 6c3f360699f6e8711c163efa3ac87948a685cd63

File: openid/src/main/java/com/inrupt/client/openid/OpenIdSession.java (hunk 2 above)
Comment (NSeydoux) [reply openid/src/main/java/com/inrupt/client/openid/OpenIdSession.java]: Fixed in cddd3e9bde469a93644c38a75b41643632055b6f

File: openid/src/main/java/com/inrupt/client/openid/OpenIdSession.java (hunk 3 above)
Comment (NSeydoux) [reply openid/src/main/java/com/inrupt/client/openid/OpenIdSession.java]: Fixed in cddd3e9bde469a93644c38a75b41643632055b6f

**Review by acoburn — APPROVED** [review]
LGTM

**Pull request merged**